pkg/activator/handler/handler.go

```diff
@@ -100,13 +100,21 @@ func withOrigProto(or *http.Request) prober.Preparer {
 	}
 }
 
-func (a *activationHandler) probeEndpoint(logger *zap.SugaredLogger, r *http.Request, target *url.URL) (bool, int) {
+func (a *activationHandler) probeEndpoint(logger *zap.SugaredLogger, r *http.Request, target *url.URL, revID activator.RevisionID) (bool, int) {
 	var (
 		attempts int
 		st       = time.Now()
 		url      = target.String()
 	)
 
+	// This opportunistically caches the probes, so
+	// a few concurrent requests might result in concurrent probes
+	// but requests coming after won't.
+	if !a.throttler.ShouldProbe(revID) {
```

Comment:
Should this move outside of the `probeEndpoint` function to not create the spans in this case?  Or move the span in? I guess I changed that recently 🙈 

I'd personally prefer to move it outside as it makes the path below more clear and tells you what's happening more explicitly.

Reply:
I wanted really the mark and should to be colocated, but I guess it's fine.